### PR TITLE
DashboardScene: Fix access to dashboard settings views

### LINKED ISSUE #80877: Dashboard settings: Settings views are accessble if the dasboard is non-editable.
Non-editable dashboards allow accessing the entire dashboard settings section. Only General settings with "Make editable" should be allowed in such a scenario.

### LINKED ISSUE #84564: Permissions tab should not appear if user does not have permission to edit those
See screenshot
Screenshot 2024-03-15 at 13 02 37

### PR DESCRIPTION
**Pull request by dprokop**
Fixes https://github.com/grafana/grafana/issues/80877
Fixes https://github.com/grafana/grafana/issues/84564

Also fixes old arch problem, when dashboard settings other than general could still be accessed via url is the dashboard was read-only

~Also, bringing a UX change to make dashboard editable once it was set to read-only.~

~In the old architecture, they were required to go to Dashboard Settings when they would have this action available:~
image

~However, this does not work in the new UX, as we now have the explicit Dashboard edit button that guards access to Dashboard settings. Hence I'm adding explicit Make editable button to the toolbar if they can make dashboard editable:~

image

### CODE REVIEW

**Comment by kaydelaney** [comment]
Noticed an empty "show more items" menu if you change to read-only and save:

https://github.com/grafana/grafana/assets/45561153/ff5aeaa7-e3f7-4375-ade8-5cca255a97e2

File: public/app/features/dashboard-scene/scene/DashboardScene.tsx [hunk 1]
```diff
@@ -354,7 +354,20 @@ export class DashboardScene extends SceneObjectBase<DashboardSceneState> {
       overlay: new SaveDashboardDrawer({
         dashboardRef: this.getRef(),
         saveAsCopy,
-        onSaveSuccess,
+        onSaveSuccess: () => {
+          onSaveSuccess?.();
+
+          // After saving we are updating meta to reflect runtime state of the edit options
```
Comment (ivanortegaalba) [inline public/app/features/dashboard-scene/scene/DashboardScene.tsx:360]: Why do we need this? It is not trivial or easy to see, maybe adding the why to the comment will help :)

File: public/app/features/dashboard-scene/scene/DashboardScene.tsx [hunk 2]
```diff
@@ -801,7 +822,7 @@ export class DashboardScene extends SceneObjectBase<DashboardSceneState> {
   canEditDashboard() {
     const { meta } = this.state;
 
-    return Boolean(meta.canEdit || meta.canMakeEditable);
+    return Boolean(meta.canEdit /*|| meta.canMakeEditable*/);
```
Comment (ivanortegaalba) [inline public/app/features/dashboard-scene/scene/DashboardScene.tsx:825]: Is this on purpose?

**Comment by dprokop** [comment]
> Noticed an empty "show more items" menu if you change to read-only and save:


Interesting, can't reproduce this... Overall, I think after making the dashboard read-only we should exit the edit mode right away. Otherwise it doesn't make sense.

**Review by ivanortegaalba — APPROVED** [review]
Looks great to me! And the UX is muuuch better that the one on the old architecture.

Just some tests here would help with any regression, since this is a user flow we don't see much as maintainers.

File: public/app/features/dashboard-scene/scene/DashboardScene.tsx (hunk 1 above)
Comment (dprokop) [reply public/app/features/dashboard-scene/scene/DashboardScene.tsx]: no need for that anymore :)

**Pull request merged**